### DISCUSSION TITLE
Need to change Docker-container MTU.

### DISCUSSION THREAD

**Discussion by ITgeevoSOC** [Security-Onion-Solutions/securityonion#6687] [Unsupported Versions] [answered]
Dear all,

Please can someone advice how can we change Docker-container MTU from 1500 to 1450 for all the interfaces?

Thanks in advance.

**Comment by tomjones1977** [comment]
To set the MTU for all containers you can add the below lines to the file `/etc/docker/daemon.json`:

```
{
  "mtu": 1454
}
```

You will need to restart to let it take effect.

**Reply by ITgeevoSOC** [reply]
Thanks for the reply.

After I configure the above line and restart the Security Onion, the configuration file goes to the default. 

Please can you advise?

**Reply by tomjones1977** [reply]
That sounds like the config file is managed by salt somewhere - perhaps @dougburks can elaborate?

**Reply by ITgeevoSOC** [reply]
Yes is manage by salt. Hopefully @dougburks can elaborate so we can resolve an issue with suricata update.

Thanks very much for your help

**Comment by dougburks** [comment, marked as answer]
I haven't tried this, but you may be able to copy `/opt/so/saltstack/default/salt/common/files/daemon.json` to `/opt/so/saltstack/local/salt/common/files/` and then make your changes there.

**Reply by ITgeevoSOC** [reply]
Thanks @dougburks for the reply. I have edited the daemon.json file and I added the MTU parameter as tomjones1977 mention above, but after I restart the server the docker is crush and stays in the starting process. 

Because I can't change the MTU from the private interface of the Hetzner hosting, is possible to change the docker MTU with another configuration command?

Thanks in advance.

**Reply by m0duspwnens** [reply]
@ITgeevoSOC Since the service is failing to start, it sounds like `/opt/so/saltstack/local/salt/common/files/daemon.json` may be malformed. Please ensure it looks like the following:

```
{%- set DOCKERRANGE = salt['pillar.get']('docker:range', '172.17.0.0/24') %}
{%- set DOCKERBIND = salt['pillar.get']('docker:bip', '172.17.0.1/24') %}
{
    "registry-mirrors": [ "https://:5000" ],
    "bip": "{{ DOCKERBIND }}",
    "default-address-pools": [
      {
        "base" : "{{ DOCKERRANGE }}",
        "size" : 24
      }
    ],
    "mtu": 1450
}
```

**Reply by ITgeevoSOC** [reply]
Dear Josh,

Thanks very much. This works. After we change the docker MTU also resolve the issue with the so-rule-update command to update the nids rule.

Thanks all for your help on this issue.
